overlord/snapstate/check_snap.go

```diff
@@ -208,16 +208,26 @@ func checkSnap(st *state.State, snapFilePath string, si *snap.SideInfo, curInfo
 		return err
 	}
 
+	snapName, instanceKey := snap.SplitInstanceName(instanceName)
+	// update instance key to what was requested
+	s.InstanceKey = instanceKey
+
 	st.Lock()
 	defer st.Unlock()
 
+	// allow registered checks to run first as they may produce more
+	// precise errors
 	for _, check := range checkSnapCallbacks {
 		err := check(st, s, curInfo, flags)
 		if err != nil {
 			return err
 		}
 	}
 
+	if snapName != s.SnapName() {
```

Comment:
do we need a test for this when instanceKey == ""

Reply:
Test added.